driver/src/price_estimation/price_source.rs

```diff
@@ -57,20 +58,45 @@ impl Token {
 
 /// An abstraction around a type that retrieves price estimate from a source
 /// such as an exchange.
-#[cfg_attr(test, mockall::automock)]
 pub trait PriceSource {
     /// Retrieve current prices relative to the OWL token for the specified
     /// tokens. The OWL token is pegged at 1 USD with 18 decimals. Returns a
     /// sparse price array as being unable to find a price is not considered an
     /// error.
-    fn get_prices(&self, tokens: &[Token]) -> Result<HashMap<TokenId, u128>>;
+    fn get_prices<'a>(
+        &'a self,
+        tokens: &'a [Token],
+    ) -> BoxFuture<'a, Result<HashMap<TokenId, u128>>>;
 }
 
 /// A no-op price source that always succeeds and finds no prices.
 pub struct NoopPriceSource;
 
 impl PriceSource for NoopPriceSource {
-    fn get_prices(&self, _: &[Token]) -> Result<HashMap<TokenId, u128>> {
-        Ok(HashMap::new())
+    fn get_prices(&self, _: &[Token]) -> BoxFuture<Result<HashMap<TokenId, u128>>> {
+        async { Ok(HashMap::new()) }.boxed()
     }
 }
+
+// We would like to tag `PriceSource` with `mockall::automock` but mockall does not support the
```

Comment:
Would it be easier for future maintenance to pass the tokens by value rather than reference?

Reply:
Reference is more correct because the value does not need to be owned. I dislike changing our real code because of limitations in the mocking framework. The mockall developer made it sound like this is something that would be fixed in the future at which we no longer need the workaround.

Reply:
I merged it already but after thinking about it again maybe the owned version is better because after all it is simpler and not in a place where performance matters. I might change it to that in another PR.

Reply:
Thanks, I'd appreciate that since this workaround is quite hacky and the developer didn't make it sound like the fix would happen in our lifetime 😉